src/SubmitPage/SubmitForm.js

```diff
@@ -341,11 +341,7 @@ class SubmitForm extends React.Component {
 
   render() {
     const { translations, location } = this.props;
-    const isUpdate =
-      location &&
-      location.state &&
-      location.state.item &&
-      location.state.item.StoreId;
+    const isUpdate = location && location.state && location.state.item;
```

Comment:
i'm not sure why storeId is not to be checked. can we name this `isUpdateForExistingStore` just to be explicit?

Reply:
Because storeId is only present when we're updating an existing store. As per the ticket we should disable editing of name and address field even when a user is coming from `Add` button of missing block component. I guess the variable name should be changed to something different.

Reply:
ah right, i mistook it for something else. yes, i think a more descriptive variable name will be helpful :)

Reply:
i updated it https://github.com/covid-maps/covid-maps/pull/307/commits/fb632a0dcee06af587b4a4a09e6e000ce3f55d6b

Reply:
👍